### PR TITLE
LPD-31975  Address ERC gaps in Headless APIs for productAccountGroups and ProductChannels

### PR DESCRIPTION
**Pull request by monaco-fabio**
https://liferay.atlassian.net/browse/LPD-31975

https://liferay.atlassian.net/browse/LPD-32510

#5332

### CODE REVIEW

**Comment by monaco-fabio** [comment]
ci:test:sf

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-test/src/testIntegration/java/com/liferay/headless/commerce/admin/catalog/resource/v1_0/test/ProductResourceTest.java [hunk 1]
```diff
@@ -66,6 +74,17 @@ public void setUp() throws Exception {
 		ServiceContext serviceContext =
 			ServiceContextTestUtil.getServiceContext(testCompany.getGroupId());
 
+		_accountGroup = _accountGroupLocalService.addAccountGroup(
+			user.getUserId(), null, RandomTestUtil.randomString(),
+			serviceContext);
+
+		_accountGroup.setDefaultAccountGroup(false);
+		_accountGroup.setType(AccountConstants.ACCOUNT_GROUP_TYPE_STATIC);
+		_accountGroup.setExpandoBridgeAttributes(serviceContext);
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-test/src/testIntegration/java/com/liferay/headless/commerce/admin/catalog/resource/v1_0/test/ProductResourceTest.java:83]: why do you need this?

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-test/src/testIntegration/java/com/liferay/headless/commerce/admin/catalog/resource/v1_0/test/ProductResourceTest.java [hunk 2]
```diff
@@ -305,6 +327,9 @@ public void testPostProduct() throws Exception {
 		assertValid(getProduct);
 
 		_testPostProductVirtual();
+
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-test/src/testIntegration/java/com/liferay/headless/commerce/admin/catalog/resource/v1_0/test/ProductResourceTest.java:330]: remove empty line

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-test/src/testIntegration/java/com/liferay/headless/commerce/admin/catalog/resource/v1_0/test/ProductResourceTest.java [hunk 3]
```diff
@@ -557,9 +582,83 @@ private void _testPostProductVirtual() throws Exception {
 		Assert.assertNotNull(productVirtualSettingsFileEntry.getSrc());
 	}
 
+	private void _testPostProductWithAccountGroupExternalReferenceCode()
+		throws Exception {
+
+		Product randomProduct = _randomProductWithSku();
+
+		randomProduct.setProductAccountGroupFilter(true);
+
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-test/src/testIntegration/java/com/liferay/headless/commerce/admin/catalog/resource/v1_0/test/ProductResourceTest.java:591]: remove empty line

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-test/src/testIntegration/java/com/liferay/headless/commerce/admin/catalog/resource/v1_0/test/ProductResourceTest.java [hunk 4]
```diff
@@ -557,9 +582,83 @@ private void _testPostProductVirtual() throws Exception {
...
+					}
+				}
+			});
+
+		Product postProduct = productResource.postProduct(randomProduct);
+
+		Product expectedProduct = postProduct.clone();
+
+		BeanTestUtil.copyProperties(randomProduct, expectedProduct);
+
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-test/src/testIntegration/java/com/liferay/headless/commerce/admin/catalog/resource/v1_0/test/ProductResourceTest.java:607]: I think not needed (you are not using these below)

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-test/src/testIntegration/java/com/liferay/headless/commerce/admin/catalog/resource/v1_0/test/ProductResourceTest.java [hunk 5]
```diff
@@ -557,9 +582,83 @@ private void _testPostProductVirtual() throws Exception {
...
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-test/src/testIntegration/java/com/liferay/headless/commerce/admin/catalog/resource/v1_0/test/ProductResourceTest.java:639]: not needed

**Review by smottal — CHANGES_REQUESTED** [review]

**Comment by monaco-fabio** [comment]
ci:test:sf

**Comment by monaco-fabio** [comment]
ci:test:sf

**Comment by monaco-fabio** [comment]
ci:test:relevant

**Comment by monaco-fabio** [comment]
ci:test:sf

**Comment by monaco-fabio** [comment]
ci:test:relevant

**Comment by smottal** [comment]
ci:test:stable

**Review by smottal — APPROVED** [review]

**Comment by smottal** [comment]
ci:report:5602760

**Comment by smottal** [comment]
ci:forward:force

**Pull request closed without merging**